api/v1/clusterextension_types.go

```diff
@@ -244,6 +243,7 @@ type CatalogFilter struct {
 	// For more information on semver, please see https://semver.org/
 	//
 	// +kubebuilder:validation:MaxLength:=64
+	// +kubebuilder:validation:MinLength:=1
```

Comment:
Breaking change because explicitly setting `version: ""` is no longer valid, but IIRC the original intention was to actually have it so you either don't provide this or explicitly provide a range so this would be more appropriate.

Ratcheting should take effect here, but again - up to maintainer discretion.